dask/array/core.py

```diff
@@ -2014,6 +2024,140 @@ def from_array(x, chunks, name=None, lock=False, asarray=True, fancy=True,
+        initialisation, if the chunking scheme in the on-disc dataset is not
+        optimal for the calculations to follow.
+    kwargs: passed to zarr's open functions.
+    """
+    import zarr
+    storage_options = storage_options or {}
+    if isinstance(url, str):
+        fs, fs_token, path = get_fs_token_paths(
+            url, 'rb', storage_options=storage_options)
+        assert len(path) == 1
+        mapper = get_mapper(fs, path[0])
+    else:
+        mapper = url
+    if component is None:
+        z = zarr.open_array(mapper, mode='r', **kwargs)
+    else:
+        z = zarr.open_group(mapper, mode='r', **kwargs)[component]
+    chunks = chunks if chunks is not None else z.chunks
+    return from_array(z, chunks, name='zarr-%s' % url)
+
+
+def to_zarr(arr, url, component=None, storage_options=None,
+            overwrite_group=False, compute=True,
+            return_stored=False, **kwargs):
+    """Save array to the zarr storage format
+
+    See https://zarr.readthedocs.io for details about the format.
+
+    Parameters
+    ----------
+    arr: dask.array
+        Data to store
+    url: str or MutableMapping
+        Location of the data. A URL can include a protocol specifier like s3://
+        for remote data. Can also be any MutableMapping instance, which should
+        be serializable if used in multiple processes.
+    component: str or None
+        If the location is a zarr group rather than an array, this is the
+        subcomponent that should be created/over-written.
+    storage_options: dict
+        Any additional parameters for the storage backend (ignored for local
+        paths)
+    overwrite_group: bool
+        If component is not none, this controls whether to create/overwrite the
+        zarr group, or if the group must exits already (but the specific
+        dataset is always overwritten)
+    compute, return_stored: see ``store()``
+    kwargs: passed to zarr's open functions, e.g., compression options
+    """
+    import zarr
+    if not _check_regular_chunks(arr.chunks):
+        raise ValueError('Attempt to save array to zarr with irregular '
+                         'chunking, please call `arr.rechunk(...)` first.')
+    storage_options = storage_options or {}
+    if isinstance(url, str):
+        fs, fs_token, path = get_fs_token_paths(
+            url, 'rb', storage_options=storage_options)
+        assert len(path) == 1
+        mapper = get_mapper(fs, path[0])
+    else:
+        # assume the object passed is already a mapper
+        mapper = url
+    chunks = [c[0] for c in arr.chunks]
+    if component is None:
+        z = zarr.open_array(
+            mapper, mode='w', shape=arr.shape, chunks=chunks, dtype=arr.dtype,
+            **kwargs)
+    else:
+        mode = 'w' if overwrite_group else 'r+'
+        z = zarr.open_group(mapper, mode=mode).create_dataset(
+            component, shape=arr.shape, chunks=chunks, dtype=arr.dtype,
+            **kwargs)
```

Comment:
I think it would be better to have an ``overwrite`` argument into this function, rather than an ``overwrite_group`` argument. 

Then this whole ``if ... else ...`` block could be replaced by:

```
z = zarr.create(shape=arr.shape, chunks=chunks, dtype=arr.dtype, store=mapper, path=component, overwrite=overwrite, **kwargs) 
```

As well as being simpler code, this also would simplify the logic around whether to overwrite existing data, because the behaviour will be the same whether or not the ``component`` argument is provided. I.e., with this change, if the user provides ``overwrite=False``, then if an array exists an exception will be raised. Conversely, if user provides ``overwrite=True``, an existing array will be deleted and overwritten.

Reply:
Yeah the open... functions are sometimes a bit of a distraction.

Reply:
Is it worth deprecating/consolidating them?

Reply:
Yes worth considering. Raised https://github.com/zarr-developers/zarr/issues/264 for discussion.